internal/store/elasticsearch/discovery_search_repository.go

```diff
@@ -237,7 +237,7 @@ func (repo *DiscoveryRepository) buildFilterExistsQueries(fields []string) ([]el
 	return filterQueries, true
 }
 
-func (repo *DiscoveryRepository) buildFunctionScoreQuery(query elastic.Query, rankBy string) elastic.Query {
+func (repo *DiscoveryRepository) buildFunctionScoreQuery(query elastic.Query, rankBy string, text string) elastic.Query {
 	if rankBy == "" {
```

Comment:
This would mean that we prioritise exact text match only if rank by is specified. Is this our intention?

Reply:
Valid point, fixed now.